design/one-pager-stacks-versioning.md

```diff
@@ -0,0 +1,248 @@
+
+#### Why standardize on semantic versions?
+
+Semantic versions are a well adopted standard that give you context just from looking at the version. Based on semantic
+ versions we could implement a variety of useful features, that we couldn't reasonably or reliably do if everyone is
+ using a unique versioning system.
+
+* Auto-upgrade a major, minor, patch version based on policy.
+* Detect if an upgrade is available based only on the tag and the latest version in a channel, no introspecting of a
+  docker package.
+* Warning if the next latest version is a major upgrade.
+* Compare versions running easily based on status of stack install.
+* Tie human readable versions to change logs.
+* I can remember a version, it's very hard to remember a sha.
+* Semantic versions used by helm and widely adopted in kubernetes community.
+
+## Design
+
+We will rely on the established standard of [semantic version 2.0](https://semver.org/spec/v2.0.0.html) embedding these
+ versions inside of the stack app.yaml metadata, and inside of the tag in a docker image.
+
+The stack app.yaml version should always be a simple stable semantic version with no pre-release metadata, and tags
+ must exist that match the semantic version in the release channel.
+
+Creation dates will be tracked in the release channel versions for auditing purposes and will be iso-8601 format.
+ This is the same format as kubernetes and is a well established standard. We will also include a sha id of the
+ underlying stack container, which is immutable, and will be used by the installer to fetch the exact artifact
+ referenced in the release channel.
+
+The stack install CRD will get a new optional field for channel in the spec, that will include the url to the channel.
+
+There will be two lists for channel information, namely the latest available in a channel, as well as the list of
+ all available versions for the stack installer to manage more complicated install scenarios based on upgrade policies
+ that will be further defined in the future.
+
+### Considerations
+
+Our underlying stack storage is a docker registry, which can tag versions of stacks in any arbitrary way, but tags must
+ point to a single sha of an actual docker image, so there is no historical context, i.e. previous stable versions.
+ Further, if you were to use a :beta tag as a stand in for a beta channel, you would have little recourse to know how to
+ downgrade to the previous beta version in the event of a problem, nor would you even know with absolute certainty which
+ version you were running by looking at a spec or status in a kubernetes resource.
+
+### Versioning
+
+Semantic versions are widely respected as a standard to version software and are well
+ understood in the software community. Helm charts rely on semantic versions, and many others. Please see examples of
+ versioning systems linked at the bottom in background research on versioning.
+
+```
```

Comment:
Nitpick: this seems more like a [block quote](https://daringfireball.net/projects/markdown/syntax#blockquote) than a code block